backend/src/apiserver/server/api_converter.go

```diff
@@ -148,14 +152,48 @@ func toApiParameters(paramsString string) ([]*api.Parameter, error) {
 	return apiParams, nil
 }
 
+func toApiRuntimeConfig(modelRuntime model.RuntimeConfig) (*api.PipelineSpec_RuntimeConfig, error) {
+	var runtimeParamsStringsMap map[string]string
+	var runtimeParams map[string]*structpb.Value
+	if modelRuntime.Parameters != "" {
+		err := json.Unmarshal([]byte(modelRuntime.Parameters), &runtimeParamsStringsMap)
+		if err != nil {
+			return nil, util.NewInternalServerError(err, fmt.Sprintf("Cannot unmarshal RuntimeConfig Parameter to map[string]string, string value: %+v", modelRuntime.Parameters))
+		}
+		runtimeParams = make(map[string]*structpb.Value)
+		for k, v := range runtimeParamsStringsMap {
+			var protoValue structpb.Value
+			err := protoValue.UnmarshalJSON([]byte(v))
+			if err != nil {
+				return nil, util.NewInternalServerError(err, fmt.Sprintf("Cannot unmarshal string %+v to structpb.Value", v))
+			}
+			runtimeParams[k] = &protoValue
```

Comment:
same here, for unmarshal, is it possible to do it in 1 step instead of unmarshal the value first then unmarshal the map?

Reply:
I don't think it was necessary. I have updated the code accordingly.